### DISCUSSION TITLE
The convolution kernel for KDP calculation

### DISCUSSION THREAD

**Discussion by Syize** [ARM-DOE/pyart#1402] [Q&A]
Hi, all. I'm puzzled by the setting of convolution kernel which is used to calculate KDP from φDP in two function: `phase_proc_lp` and `phase_proc_lp_gf` in `pyart.correct.phase_proc`.

Here is a part of the code
```
# prepare output
sobel = 2.0 * np.arange(window_len) / (window_len - 1.0) - 1.0
sobel = sobel / (abs(sobel).sum())
sobel = sobel[::-1]
gate_spacing = (radar.range["data"][1] - radar.range["data"][0]) / 1000.0
kdp = scipy.ndimage.convolve1d(proc_ph["data"], sobel, axis=1) / (
    (window_len / 3.0) * 2.0 * gate_spacing
)
```

I thought that the convolution kernel for KDP calculation is `[-0.2, -0.1, 0, 0.1, 0.2]` too. Is there any relevant literature for reference?

**Comment by kmuehlbauer** [comment]
A very comprehensive paper about this kind of differentiation is: https://arxiv.org/pdf/1102.5219

Another resource, where the actual variant is described in more detail: http://www.holoborodko.com/pavel/numerical-methods/numerical-derivative/lanczos-low-noise-differentiators/

Not sure if this actually answers your question, but those resources describe the mathematical background of this derivation by convolution.
